hack/internal/test-images/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM python:3.12-slim
+
+# Set environment variable to avoid interactive prompts during package installation
+ENV DEBIAN_FRONTEND=noninteractive
+ENV RAY_VERSION=2.41.0
```

Comment:
I think it's much better to use a build argument to change the version in a single place.